token/core/common/metrics/metrics.go

```diff
@@ -0,0 +1,138 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package metrics
+
+import (
+	"time"
+
+	"github.com/hyperledger-labs/fabric-smart-client/platform/view/services/metrics"
+	"github.com/hyperledger-labs/fabric-token-sdk/token"
+)
+
+type (
+	CounterOpts   = metrics.CounterOpts
+	Counter       = metrics.Counter
+	GaugeOpts     = metrics.GaugeOpts
+	Gauge         = metrics.Gauge
+	HistogramOpts = metrics.HistogramOpts
+	Histogram     = metrics.Histogram
+	Provider      = metrics.Provider
+)
+
+var GetProvider = metrics.GetProvider
+
+type Metrics struct {
+	provider Provider
+	labels   []string
+
+	Issues        Counter
+	FailedIssues  Counter
+	IssueDuration Histogram
+
+	Transfers        Counter
+	FailedTransfers  Counter
+	TransferDuration Histogram
+
+	Audits        Counter
+	FailedAudits  Counter
+	AuditDuration Histogram
+}
+
+func New(provider Provider, tmsID token.TMSID) *Metrics {
+	m := &Metrics{
+		provider: provider,
+		labels: []string{
+			"network", tmsID.Network,
+			"channel", tmsID.Channel,
+			"namespace", tmsID.Namespace,
+		},
+	}
+	m.Issues = m.NewCounter(issuesOpts)
+	m.FailedIssues = m.NewCounter(failedIssuesOpts)
+	m.IssueDuration = m.NewHistogram(issueDurationOpts)
+
+	m.Transfers = m.NewCounter(transfersOpts)
+	m.FailedTransfers = m.NewCounter(failedTransfersOpts)
+	m.TransferDuration = m.NewHistogram(transferDurationOpts)
+
+	m.Audits = m.NewCounter(auditsOpts)
+	m.FailedAudits = m.NewCounter(failedAuditsOpts)
+	m.AuditDuration = m.NewHistogram(auditDurationOpts)
+	return m
+}
+
+func (m *Metrics) AddIssue(tokenType string, noErr bool) {
+	if noErr {
+		m.Issues.With("token_type", tokenType).Add(1)
+		return
+	}
+	m.FailedIssues.With("token_type", tokenType).Add(1)
+}
+
+func (m *Metrics) ObserveIssueDuration(duration time.Duration) {
+	m.IssueDuration.Observe(float64(duration.Milliseconds()))
+}
+
+func (m *Metrics) AddTransfer(noErr bool) {
```

Comment:
I think it would be easier to understand if we used two labels instead. So instead of `m.AddIssue("CHF", err == nil)` we could do
```
m.Issues.With("token_type", tokenType, "failed", err != nil).Add(1)
```